benchmarks/hash_table/static_map/find_bench.cu

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <defaults.hpp>
+#include <key_generator.hpp>
+
+#include <cuco/static_map.cuh>
+
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/iterator/discard_iterator.h>
+
+namespace cuco::benchmark {
+/**
+ * @brief A benchmark evaluating `find` performance:
+ * - Total number of insertions: 100'000'000
+ */
+template <typename Key, typename Value, typename Dist>
+std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> nvbench_static_map_find(
+  nvbench::state& state, nvbench::type_list<Key, Value, Dist>)
+{
+  auto const num_keys      = state.get_int64("NumInputs");
+  auto const occupancy     = state.get_float64("Occupancy");
+  auto const matching_rate = state.get_float64("MatchingRate");
+
+  std::size_t const size = num_keys / occupancy;
+
+  thrust::device_vector<Key> keys(num_keys);
+
+  key_generator gen;
+  gen.generate<Dist>(state, thrust::device, keys.begin(), keys.end());
+
+  auto pairs_begin = thrust::make_transform_iterator(
+    keys.begin(), [] __device__(auto i) { return cuco::pair_type<Key, Value>(i, i); });
```

Comment:
I'm still scratching my head over if using a `transform_iterator` over `transform` is correct here. With the iterator approach, we only load the keys from gmem and construct the pairs ad-hoc in the kernel. This means we actually load `sizeof(key_type)*N` from memory instead of loading `sizeof(pair_type)*N` bytes. My experiments show that there's not much difference in timing. So maybe this discussion is just bikeshedding.

Reply:
For query actions like `find`, `count`, this is not an issue since `pairs` are used to insert thus using iterators or not won't affect the benchmark results. For `insert`, how important is this memory load compared to the overall runtime?

Reply:
> For insert, how important is this memory load compared to the overall runtime?

It's not affecting the overall runtime that much (<2%). 

The question is if we're bending the  rules of the benchmark setup with this.

Reply:
Technically the memory load we are talking about happens at
```cpp
auto pair = *(pairs_begin + i);
```
Thus it's loading `sizeof(pair_type)` whatsoever, right?

Reply:
If I understand correctly how `thrust::transform_iterator` works, then the actual magic happens inside the `operator*` here. Inside the operator, we load a key from gmem and construct a new pair from it, which we then return. So we're just _loading_ a key instead of a key-value pair from gmem.